Assistants lair vault

Probe1 opened the pull request:
Adds a fun little maze vault for players to explore.  No pictures because no spoilers.  It's a fun exploration experience so the loot mid tier and the danger is low.

I hope you guys enjoy it!

<!--
Pull requests must be atomic.  Change one set of related things at a time.  Bundling sucks for everyone.
This means, primarily, that you shouldn't fix bugs and add content in the same PR. When we mean 'bundling', we mean making one PR for multiple, unrelated changes.

Test your changes. PRs that do not compile will not be accepted.
Testing your changes locally is incredibly important. If you break the serb we will be very upset with you.

Large changes require discussion.  If you're doing a large, game-changing modification, or a new layout for something, discussion with the community is required as of 26/6/2014.  Map and sprite changes require pictures of before and after.  MAINTAINERS ARE NOT IMMUNE TO THIS.  GET YOUR ASS IN IRC.

Merging your own PRs is considered bad practice, as it generally means you bypass peer review, which is a core part of how we develop.

It is also suggested that you hop into irc.rizon.net #vgstation to discuss your changes, or if you need help.

When working with in body changelogs, the syntax is as follows:
:cl:
 * rscadd: Did stuff!
 * rscdel: did other stuff!

for the keys you can use in a changelog, they are the same as described in html/changelogs/example.yml
NOTE that anything *after* the :cl: will be parsed as a changelog, if it somehow manages to be parseable as such, so always put the changelog at the VERY end!
-->

🆑 

* rscadd: Fun little maze vault for you to explore.

HarseTheef commented:
It's not very honest or atomic to wordlessly rebalance how mines work, and then include some in the vault.  As well as the inclusion of admin-spawn-only stuff, and state it is a secret.

Talvan commented:
It's secret just merge it guys :^)

Probe1 commented:
He updated a 9 year old proc to use something that works much better to do the same thing!  Get em boys!

HarseTheef commented:
> He updated a 9 year old proc to use something that works much better to do the same thing! Get em boys!

Why do you continue to be intentionally dishonest.

https://dl.dropboxusercontent.com/s/t5xo8xxrfefr2qx/2018-03-01_17-10-12.gif

This is the original functionality.  As you can see, it is an exactly 60 real second stun in which you retain the items in your hands and you are not affected by ZAS given that magboots will remain engaged while you stand.

https://dl.dropboxusercontent.com/s/t5xo8xxrfefr2qx/2018-03-01_17-10-12.gif

This is after your change.  You dropped it from 60 seconds to 20 seconds.  That is a 40 second difference.  Yours is a knockdown which will cause items to both be dropped from your hands and for magboots to be rendered useless, as you will be affected by ZAS even when laying down while they are engaged.

Not only are the effects not functionally identical, you but you made it only 1/3rd the original duration.

Probe1 commented:
All you had to say was increase knockdown(10) to (30).  I don't know where the sperg out comes from.

Aurx commented:
It comes from you making balance changes with no changelog, duh.

SonixApache commented on code/game/objects/effects/mines.dm:
```diff
@@ -77,6 +77,6 @@
 /obj/effect/mine/stun/trigger(mob/AM)
 	if(ismob(AM))
-		AM.Stun(30)
+		AM.Knockdown(10)
```
just make a new type, man

SonixApache commented:
i for one welcome my dylan overlords

Probe1 commented:
Ah you got me there I go rebalancing orphaned items that I think only I remember exist.  

No for real I've been told you have a massive hateboner for me and that's fine, you're welcome to hate me.  I can't remember _why_ you hate me but I've been told I did something as an admin and it upset you.  But let's focus that into something positive.  For example:  mines have no attackby() so there's no way to pick them up or manipulate them.  The only way to avoid triggering them is to not touch them or bring a guinea pig along to touch them first.  It's old shitcode of the highest order.  I'll come back to that.  Second and more importantly, the file is FULL of ancient code that needs refactored.   ``` "<font color='red'> ``` instead of <span class = warn> ``` .  Probably more that I haven't noticed.  

Now back to the caveat I wanted to expand on.  Mines are an abandoned item that to my knowledge don't exist on any spawner, any map, any list.  They can only be called to existence by an admin and I only know about them because I found them while experimenting as a new admin in 2015.  As far as I know they've never existed outside of their file on /vg/ and are an orphaned item that has miraculously lived in our code for 7 years so I think they're neat and it'd be fun to toss them into a vault.

To wrap all this up, launching into a pissy tirade every time I do something isn't very productive and you don't have to dial it up to 10 for me to notice you.  If you think these stun mines should knock out players for a full _sixty_ seconds and I'm destroying /vg/ by altering that go ahead and tell me.  But be reasonable about it.  Getting super mad that I copied some code another collab gave me in Discord when I asked what I should update Stun() to isn't helping you, me, or the server.

SonixApache commented:
i don't feel like checking everything, but github's shitty search function already shows one other vault with a stun mine

balance changes, even if they're for the better, shouldn't be snuck in with mapping changes *without notifying anyone* -- this is a standard that's been set up a while ago because of certain people doing this every time


in short, stop being so pissy and just put it in the changelog (or add a mine subtype, or make a seperate PR for the changes, etcetera, etcetera)

Probe1 commented:
Tell you what I'll split the difference and add a balance tag until I up the timer on them when I fix the conflicts and the rest of the files outdated code.

Dacendeth commented:
That broken engineering vault with a spacepod fabricator has that mine, it's super annoying and I always forget it's there.

Probe1 commented:
Yeah that ones on me.  The only other place this puppy exists in the code is on that vault and I added it _without testing_.  This time, because the vault has a lot more, I tested it and realized they're dogshit and needed updated.  So I changed it to not be terribly and here we are in Ultimate Bike Shed 2018.

Unknown-as-Captain commented:
Hahahahahahahaha How The Fuck Is Grudging Real Hahahaha Nigga Just Follow The Standards Like Nigga Document Your Changes Haha

DamianX commented:
Meme mine has been restored to its shitcode glory, rejoice.

Unknown-as-Captain commented:
Nobody wanted that.
Just don't be sneaky or you'll get called Dylan.

SonixApache commented:
Why am I loading this now? 
![image](https://user-images.githubusercontent.com/8468269/36916482-4d449fd6-1e54-11e8-95fa-fc8a06ebed58.png)

SonixApache commented on vgstation13.dme:
```diff
@@ -2291,6 +2291,7 @@
 #include "maps\randomvaults\dance_revolution.dm"
 #include "maps\randomvaults\objects.dm"
 #include "maps\randomvaults\spessmart.dm"
+#include "maps\randomvaults\assistantslair.dmm"
```
🤔

Probe1 replied:
/shrug

I unticked it and pushed the changes.  My bad.
